lib/new_relic/agent/monitors/synthetics_monitor.rb

```diff
@@ -27,7 +29,16 @@ def on_before_call(request) # THREAD_LOCAL_ACCESS
 
         txn = Tracer.current_transaction
         txn.raw_synthetics_header = encoded_header
+        txn.raw_synthetics_info_header = info_header
         txn.synthetics_payload = incoming_payload
+        txn.synthetics_info_payload = load_json(info_header, SYNTHETICS_INFO_HEADER_KEY)
+      end
+
+      def load_json(header, key)
+        ::JSON.load(header)
+      rescue => err
+        NewRelic::Agent.logger.debug("Failure loading json header '#{key}' in #{self.class}, #{err.class}, #{err.message}")
```

Comment:
How likely is it to have invalid JSON reach this rescue? Would a `warn` or `error` level log statement be better?

Reply:
Probably not super likely, since we'll be taking it right out of the header. Also, I don't think this merits a warn or error level log though. Warn and error level both would be logged by default, unlike debug. And I don't think this information is something that could be helpful the the default logs since this only would be relevant to synthetics events, not the agent functioning in general.

Reply:
Gotcha, sounds good.